public/components/common/hooks/useEsSearch.js

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react';
```

Comment:
the name of the files must be in kebab-case `use-es-search` and all new files must be in typescript `ts` or `tsx`